### PR TITLE
Fix error output when an error is encountered in parsing file

### PR DESCRIPTION
**Pull request by pysysops**
This PR fixes the issue reported in #173 

Thought I'd have a look at this issue to learn more about the way in which goss loads configuration. Initially, it was a little confusing but I realised the reason for readJSONData in store.go is to handle configuration passed to STDIN.

The simplest answer seemed to be to amend validate.go and use readJSONData only when reading from STDIN and to make use of readJSON (which sets the StoreFormat based on file type) when passing a file config to goss.

It was also necessary to amend the setStoreFormatFromData function as the use of log.Fatalf was causing goss to exit before functions that output informative errors had run. 

 Examples of output from a `\.` in configuration:

JSON:
```
2016/11/20 22:51:23 Unable to determine format from content
Error: invalid character '.' in string escape code
```
 
YAML:
```
2016/11/20 22:49:07 Unable to determine format from content
Error: yaml: line 5: found unknown escape character
```

If configuration sent in via STDIN is invalid goss will exit with:
```
Error: StoreFormat unset
2016/11/20 23:36:27 Unable to determine format from content
```

Any feedback appreciated!

Thanks,

Tim.

### CODE REVIEW

File: store.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func setStoreFormatFromData(data []byte) {
 		StoreFormat = YAML
 		return
 	}
-	log.Fatalf("Unable to determine format from content")
+	log.Printf("Unable to determine format from content")
```
Comment (aelsabbahy) [inline store.go]: I'm thinking if we revert this back to Fatalf and update readJSONData to be:
```go
func ReadJSONData(data []byte) GossConfig {
    setStoreFormatFromData(data)
    if StoreFormat == UNSET {
        setStoreFormatFromData(data)
    }
   ...
```

The user will get clear error messages in all scenarios, what do you think?

Comment (pysysops) [inline store.go]: That sounds good, I'll try it out and test out autoadd'ing resources in both formats / replicating #157 issues.

**Comment by pysysops** [comment]
Okay for #157 I think maybe informing the user that the resulting config file would be empty and that it can't be written. It doesn't exit with an error (although it could).

I've added some basic tests to validate that an empty file isn't created or destroyed by auto-adding a _nosuchresource_ to a file with resources in.

Of course there's a few options: Create an empty file... Or an empty object in JSON?

I'm not really sure what the right answer is here so some feedback / thoughts would be appreciated.

**Comment by aelsabbahy** [comment]
Tested this, works great, just needs to be run through `goimports` then It's ready to be merged.

**Comment by pysysops** [comment]
Darn Atom and it's 2 spaces... I now have the go-imports package installed and working nicely. Thanks!! 😃

**Pull request merged**